packages/astro/test/i18n-routing.test.js

```diff
@@ -701,6 +709,19 @@ describe('[SSG] i18n routing', () => {
 				return true;
 			}
 		});
+
+		it('should only match the default locale string in URLs where it is actually used as a locale', async () => {
+			// It should _not_ match `/en` from `/spanish/enigma` 
+			// and should respond with a 200
+			let html = ''
+			try {
+				html = await fixture.readFile('/spanish/enigma/index.html');
+			} catch {
+				// catch block just here so the test can fail cleanly
+			}
+			let $ = cheerio.load(html);
+			expect($('body').text()).includes('Astro Enigma');
```

Comment:
I believe you'd want to add an assertion for "Current Locale: es", not "Astro Enigma"

Reply:
Trying to do this has revealed that the `Astro.currentLocale` is sometimes not set correctly (tested on main just now, so this includes #9998).

Example: adding

```js
expect($('body').text()).includes('Current Locale: es');
```

to a test that loads a spanish page results in:

```bash
  1) [SSG] i18n routing
       i18n routing with routing strategy [prefix-other-locales]
         should render localised page correctly when it has codes+path:
     AssertionError: expected '\nEspanol\nCurrent Locale: en ' to include 'Current Locale: es'
      at Context.<anonymous> (file:///Users/espy/Sites/sandbox/astro/packages/astro/test/i18n-routing.test.js:677:29)
```

My PR and its tests are only concerned whether the requested page is loaded or erroneously `404`ed by the i18n middleware, not whether the `currentLocale` is set correctly. Would you accept my original tests, since they do that, and we can worry about `currentLocale` in a separate issue?